api/src/utils/keycloak-utils.ts

```diff
@@ -1,40 +1,112 @@
+import { z } from 'zod';
 import { SYSTEM_IDENTITY_SOURCE } from '../constants/database';
 
 /**
- * Parses out the user's GUID from a keycloak token, which is extracted from the
- * `preferred_username` property.
+ * User information from a verified IDIR Keycloak token.
+ */
+export const IdirUserInformation = z.object({
+  idir_user_guid: z.string().toLowerCase(),
+  identity_provider: z.literal(SYSTEM_IDENTITY_SOURCE.IDIR.toLowerCase()),
+  idir_username: z.string().toLowerCase(),
+  email_verified: z.boolean(),
+  name: z.string(),
+  preferred_username: z.string(),
+  display_name: z.string(),
+  given_name: z.string(),
+  family_name: z.string(),
+  email: z.string()
+});
+export type IdirUserInformation = z.infer<typeof IdirUserInformation>;
+
+/**
+ * User information from a verified BCeID Basic Keycloak token.
+ */
+export const BceidBasicUserInformation = z.object({
```

Comment:
Any reason we don't take advantage of `v.extend`?

Reply:
Umm, not in particular. I was just thinking it is a bit less hard to read if its all just spelled out plainly. Extending really just saves a few lines of code, in this 1 file, and wouldn't impact how anyone uses them elsewhere.

Reply:
Probably a bit subjective really. I know my original keycloak types were extending one another, but it might be best just to keep things simple in general. The zod stuff adds another layer of complexity too, having both a zod object and the type.